### PR TITLE
move WaitForCacheSync to the sharedInformer package

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: part of #81172

**Special notes for your reviewer**:

/assign @liggitt @dims @andrewsykim 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/cache/shared_informer.go [hunk 1]
```diff
@@ -190,6 +190,21 @@ const (
 	initialBufferSize = 1024
 )
 
+// WaitForCacheSyncWrapped is a wrapper around cache.WaitForCacheSyncWrapped that generates log messages
```
Comment (yliaog) [inline staging/src/k8s.io/client-go/tools/cache/shared_informer.go]: the comment above needs to be revised

File: staging/src/k8s.io/client-go/tools/cache/shared_informer.go [hunk 2]
```diff
@@ -190,6 +190,21 @@ const (
 	initialBufferSize = 1024
 )
 
+// WaitForCacheSyncWrapped is a wrapper around WaitForCacheSync that generates log messages
+// indicating that the controller identified by controllerName is waiting for syncs, followed by
+// either a successful or failed sync.
+func WaitForCacheSyncWrapped(controllerName string, stopCh <-chan struct{}, cacheSyncs ...InformerSynced) bool {
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/tools/cache/shared_informer.go]: suggest `WaitForNamedCacheSync(name string, ...`

File: staging/src/k8s.io/client-go/tools/cache/shared_informer.go [hunk 3]
```diff
@@ -190,6 +190,21 @@ const (
...
+	klog.Infof("Waiting for caches to sync for %s", controllerName)
+
+	if !WaitForCacheSync(stopCh, cacheSyncs...) {
+		utilruntime.HandleError(fmt.Errorf("unable to sync caches for %s", controllerName))
+		return false
+	}
+
+	klog.Infof("Caches are synced for %s controller", controllerName)
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/tools/cache/shared_informer.go]: make this general purpose (omit "controller" from the log message)

File: staging/src/k8s.io/client-go/tools/cache/shared_informer.go [hunk 4]
```diff
@@ -190,6 +190,21 @@ const (
...
+	return true
+}
+
 // WaitForCacheSync waits for caches to populate.  It returns true if it was successful, false
 // if the controller should shutdown
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/tools/cache/shared_informer.go]: I'd add a comment to this method indicating callers should generally prefer `WaitForNamedCacheSync`

**Pull request merged**